video-app/src/app/library/home/home.component.html

```diff
@@ -1,79 +1,168 @@
+      name="list"
+      value="list"
+      aria-label="List mode"
+    >
+      List view
+    </mat-button-toggle>
+  </mat-button-toggle-group>
 
-<mat-button-toggle-group #showFav="matButtonToggleGroup" value="allVideos" class="margin-bottom margin-left"
-  (change)="onFavoriteShow(showFav)">
-  <mat-button-toggle class="mat-raised-button" id="allVideos" name="allVideos" value="allVideos" aria-label="allVideos mode">
-    All videos
-  </mat-button-toggle>
-  <mat-button-toggle class="mat-raised-button" id="favorite" name="favorite" value="favorite" aria-label="favorite mode">
-    Show favorite only
-  </mat-button-toggle>
-</mat-button-toggle-group>
+  <mat-button-toggle-group
+    #showFav="matButtonToggleGroup"
+    value="allVideos"
+    class="margin-bottom margin-left"
+    (change)="onFavoriteShow(showFav.value !== 'allVideos')"
+  >
+    <mat-button-toggle
+      class="mat-raised-button"
+      id="allVideos"
+      name="allVideos"
+      value="allVideos"
+      aria-label="allVideos mode"
+    >
+      All videos
+    </mat-button-toggle>
+    <mat-button-toggle
+      class="mat-raised-button"
+      id="favorite"
+      name="favorite"
+      value="favorite"
+      aria-label="favorite mode"
+    >
+      Show favorite only
+    </mat-button-toggle>
+  </mat-button-toggle-group>
 
-<mat-button-toggle-group #sortButton="matButtonToggleGroup" value="asc" class="margin-bottom margin-left"
-  (change)="onSortClick(sortButton)">
-  <mat-button-toggle class="mat-raised-button" id="asc" name="asc" value="asc" aria-label="asc mode">
-    Sort by the newest
-  </mat-button-toggle>
-  <mat-button-toggle class="mat-raised-button" id="desc" name="desc" value="desc" aria-label="desc mode">
-    Sort by the oldest
-  </mat-button-toggle>
-</mat-button-toggle-group>
+  <mat-button-toggle-group
+    #sort="matButtonToggleGroup"
+    value="asc"
+    class="margin-bottom margin-left"
+    (change)="onSortClick(sort.value)"
+  >
+    <mat-button-toggle
+      class="mat-raised-button"
+      id="asc"
+      name="asc"
+      value="asc"
+      aria-label="asc mode"
+    >
+      Sort by the newest
+    </mat-button-toggle>
+    <mat-button-toggle
+      class="mat-raised-button"
+      id="desc"
+      name="desc"
+      value="desc"
+      aria-label="desc mode"
+    >
+      Sort by the oldest
+    </mat-button-toggle>
+  </mat-button-toggle-group>
 </div>
 
 <div class="video-container">
-<div *ngFor="let video of videos" class="item box-corners box-border padding-around video-box-height margin-bottom"
-  [ngClass]="{'item1': viewModeValue === 'metro', 'item2': viewModeValue === 'list'}">
-  <img class="cursor-pointer video-thumbnail" src="{{video.image}}" (click)="onPlayClick(video)">
-  {{video.title}}
-  <br />
-  {{video.source}}
-  <br />
-  {{video.date | date: 'medium'}}
-  <br />
-  Liked {{video.likesCount}}, Played {{video.playsCountDescription}}
-  <br />
-  Added to your video {{video.creationDate | date: 'medium'}}
+  <div
```

Comment:
to tez powinien być osobny komponent

Reply:
cały ten div przenieść do osobnego komponentu, tak? czy tylko zaznaczone?